File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/evm/smart-accounts/smart_account.sign_typed_data.py
# Usage: uv run python evm/smart-accounts/smart_account.sign_typed_data.py

import asyncio

from cdp import CdpClient
import dotenv

dotenv.load_dotenv()


async def main():
    async with CdpClient() as cdp:
        owner = await cdp.evm.get_or_create_account(name="SignTypedData-Example-Owner")
        smart_account = await cdp.evm.get_or_create_smart_account(
            name="SignTypedData-Example-SmartAccount",
            owner=owner,
        )

        domain = {
            "name": "EIP712Domain",
            "chainId": 84532,
            "verifyingContract": "0x0000000000000000000000000000000000000000",
        }

        types = {
            "EIP712Domain": [
                {"name": "name", "type": "string"},
                {"name": "chainId", "type": "uint256"},
                {"name": "verifyingContract", "type": "address"},
            ],
        }
        primary_type = "EIP712Domain"
        message = {
            "name": "EIP712Domain",
            "chainId": 84532,
            "verifyingContract": "0x0000000000000000000000000000000000000000",
        }

        signature = await smart_account.sign_typed_data(
            domain=domain,
            types=types,
            primary_type=primary_type,
            message=message,
            network="base-sepolia",
        )
        print("Signature: ", signature)


asyncio.run(main())

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/evm/smart-accounts/smartAccount.signTypedData.ts
// Usage: pnpm tsx evm/smart-accounts/smartAccount.signTypedData.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

const owner = await cdp.evm.getOrCreateAccount({
  name: "SignTypedData-Example-Owner",
});

const smartAccount = await cdp.evm.getOrCreateSmartAccount({
  owner,
  name: "SignTypedData-Example-SmartAccount",
});

console.log("Created smart account:", smartAccount.address);

const signature = await smartAccount.signTypedData({
  domain: {
    name: "Test",
    chainId: 84532,
    verifyingContract: "0x0000000000000000000000000000000000000000",
  },
  types: {
    Test: [{ name: "name", type: "string" }],
  },
  primaryType: "Test",
  message: {
    name: "John Doe",
  },
  network: "base-sepolia",
});

console.log("Signature:", signature);
